eo-maven-plugin/src/main/java/org/eolang/maven/DiscoverMojo.java

```diff
@@ -143,12 +144,31 @@ private static Set<String> names(final XML xml) {
                         " and @base != '$'",
                         " and @base != '&'",
                         " and not(@ref)",
-                        "]/string-join((@base, @ver),'",
+                        "]/string-join((@base,",
+                        this.version(xml),
+                        "),'",
                         OnReplaced.DELIMITER,
                         "')"
                     )
                 )
             )
         );
     }
+
+    /**
+     * Get a version to concatenate with.
+     * @param xml XML
+     * @return Version to concatenate with
+     */
+    private String version(final XML xml) {
```

Comment:
@maxonfjvipon 
1. Why not? 
2. JS? What do you mean? Maybe we have some misunderstanding here. I suggest to keep `string-join` method the same as it was before that change. You can retrieve the list of names and then add a default version for unversioned names in Java only. And you will be able to avoid complex expressions in `xslt`.  What do you think?

Reply:
@volodya-lombrozo 
1. because when we write `text|0.0.2`, here `0.0.2` is the version of our `eo-runtime`, not of the `eo-strings` package. By this tag we go to the specific `Objectionary` where all objects are stored and it versions of their specific packages don't matter. You can go to check `Objectionary` by tags `0.28.5`, `0.28.6` and you'll see that all objects from `eo-strings` package has `+version 0.0.4` (for example) in both objectionaries.
2. Now I understand what you meant, I'll try

Reply:
@maxonfjvipon You are right, thank you for the detailed explanation.